ATAPHtmlReport/report.js

```diff
@@ -235,73 +235,45 @@ function calcDotPosition(){
 
 }
 
-const categoryElements = document.getElementsByClassName('orgMeasure');
-function hideMitreTechniques(checkbox){
+/*
+techniques are hidden or shown based on the status of the provided checkboxes and classes 
+classes must be in a compatible format for document.querySelectorAll()
+*/
+let activeFilter = new Array();
+function hideMitreTechniques(checkbox, classes){
+    let classElements = document.querySelectorAll(classes);
     if (checkbox.checked) {
-        for (let i = 0; i < categoryElements.length; i++) {
-            categoryElements[i].style.padding = '0.1em';
+        /* push the current classes into the activeFilter array to determine which filters are currently active. */
+        activeFilter.push(classes);
+        for (let i = 0; i < classElements.length; i++) {
+            classElements[i].style.padding = '0.1em';
 
-            const children = categoryElements[i].querySelectorAll('*');
+            const children = classElements[i].querySelectorAll('*');
             for (let j = 0; j < children.length; j++) {
                 children[j].style.display = 'none';
             }
         }
-    } else {
-        for (let i = 0; i < categoryElements.length; i++) {
-            categoryElements[i].style.removeProperty('padding');
-
-            const children = categoryElements[i].querySelectorAll('*');
-            for (let j = 0; j < children.length; j++) {
-                children[j].style.removeProperty('display'); 
-            }
+    } 
+    else {
+        activeFilter.splice(activeFilter.indexOf(classes),1);
+        /* create an array from the classElements since it makes filtering easier. */
+        let elementsToHide = Array.from(classElements);
+        let elementsNotToHide;
+        /* create an array that includes all elements from the remaining active filters */
```

Comment:
Du könntest die Zeilen ab "let ElementsNotToHide;" bis zum Ende des if sehr gut vereinfachen. 
Mein Vorschlag wäre das Ganze in eine Zeile zu packen und einen Kommentar drüber zu schreiben: 
let elementsNotToHide = (activeFilter.length === 0) ? new Array() : Array.from(document.querySelectorAll(activeFilter));